scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -741,6 +741,7 @@ private final Connection getAliveConnection(InetSocketAddress peerAddress, boole
 			} else if (!connection.isExecuting() && running.get()) {
 				LOGGER.debug("Recreate new connection for {}", connection);
 				newConnection = new Connection(connection, new SerialExecutor(executor));
+				connectionStore.remove(connection, false);
```

Comment:
This is not directly linked to the PR ? It's is a missing part of f61c070d814acab9374c210d1637ea0f50c71e9f ?

Reply:
May be.
I mainly worked and tested on that "all together" and the split it with the intention to merge it together again. I didn't ensure, that the single commits are deeply tested on their own.